src/core/atomicdex/utilities/cpprestsdk.utilities.cpp

```diff
@@ -54,3 +54,23 @@ handle_exception_pplx_task(pplx::task<void> previous_task)
 #endif
     }
 }
+
+pplx::task<std::filesystem::path> download_file(const t_http_client_ptr& client, const std::string& url, const std::filesystem::path& output_location)
+{
+    SPDLOG_DEBUG("Start downloading {} to {}", url, output_location);
+    return client->request(web::http::methods::GET, url)
+        .then([](web::http::http_response response)
+        {
+            return response.body();
+        })
+        .then([url, output_location](Concurrency::streams::istream is)
+        {
+            Concurrency::streams::streambuf<std::uint8_t> rwbuf =
+                Concurrency::streams::file_buffer<std::uint8_t>::open(output_location).get();
+            
+            is.read_to_end(rwbuf).get();
+            rwbuf.close();
+            SPDLOG_DEBUG("Finished downloading {} to {}", url, output_location);
+            return output_location;
+        });
```

Comment:
There is no `.then(&handle_pplx_task_error)` or custom error functor in case of error.